libnavigation-core/src/test/java/com/mapbox/navigation/core/preview/RoutesPreviewControllerTest.kt

```diff
@@ -0,0 +1,404 @@
+package com.mapbox.navigation.core.preview
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.internal.route.nativeRoute
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.testing.factories.createDirectionsResponse
+import com.mapbox.navigation.testing.factories.createDirectionsRoute
+import com.mapbox.navigation.testing.factories.createNavigationRoutes
+import com.mapbox.navigator.RouteAlternative
+import com.mapbox.navigator.RouteInterface
+import com.mapbox.navigator.RoutesData
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.CompletableDeferred
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Test
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class RoutesPreviewControllerTest {
+
+    @Test
+    fun `default state`() {
+        val routesPreviewController = createRoutePreviewController()
+
+        var previewUpdate: RoutesPreviewUpdate? = null
+        routesPreviewController.registerRoutesPreviewObserver {
+            previewUpdate = it
+        }
+
+        assertNull(previewUpdate)
+    }
+
+    @Test
+    fun `set previewed routes`() {
+        val routesPreviewController = createRoutePreviewController()
+        var previewUpdate: RoutesPreviewUpdate? = null
+        routesPreviewController.registerRoutesPreviewObserver {
+            previewUpdate = it
+        }
+
+        val testRoutes = createNavigationRoutes(
+            response = createDirectionsResponse(
+                uuid = "test",
+                routes = listOf(
+                    createDirectionsRoute(),
+                    createDirectionsRoute()
+                )
+            )
+        )
+        routesPreviewController.previewNavigationRoutes(testRoutes)
```

Comment:
Do you have a test that checks what the last update is passed to observer on registration?

Reply:
added